### PR TITLE
Update roadmap documentation

### PR DESCRIPTION
**Pull request by paraseba**
Split between pre and post 1.0 roadmaps.

### CODE REVIEW

File: docs/docs/contributing.md [hunk 1]
```diff
@@ -61,42 +59,80 @@ TODO
...
+#### Core format
 
 The core format is where we’ve put most of our effort to date and we plan to continue work in this area. Leading up to the 1.0 release, we will be focused on stabilizing data structures for snapshots, chunk manifests, attribute files and references. We’ll also document and add more mechanisms for on-disk format evolution. The intention is to guarantee that any new version of Icechunk can always read repositories generated with any previous versions. We expect to evolve the [spec](https://icechunk.io/spec/) and the Rust implementation as we stabilize things.
 
-### Optimizations
+#### Features
+
+- Commit conflict detection, resolution and rebase
+- Current session status (git status)
+- Support GCP
```
Comment (rabernat) [inline docs/docs/contributing.md:72]:
```suggestion
- Support Google Cloud Storage
- Support Azure Blob Storage
```

Azure is important to Microsoft and therefore Dev Seed.

File: docs/docs/contributing.md [hunk 2]
```diff
@@ -61,42 +59,80 @@ TODO
...
+- Distributed write support with dask.array
+- Credential sets for virtual datasets
+- Complete Python API
```
Comment (rabernat) [inline docs/docs/contributing.md:75]: Could we make this more specific?

**Review by rabernat — APPROVED** [review]
This is great Seba!

Where appropriate, please link explicitly to existing issues / PRs.

File: docs/docs/contributing.md [hunk 3]
```diff
@@ -61,42 +59,80 @@ TODO
...
 
 The core format is where we’ve put most of our effort to date and we plan to continue work in this area. Leading up to the 1.0 release, we will be focused on stabilizing data structures for snapshots, chunk manifests, attribute files and references. We’ll also document and add more mechanisms for on-disk format evolution. The intention is to guarantee that any new version of Icechunk can always read repositories generated with any previous versions. We expect to evolve the [spec](https://icechunk.io/spec/) and the Rust implementation as we stabilize things.
 
-### Optimizations
+#### Features
+
+- Commit conflict detection, resolution and rebase
+- Current session status (git status)
+- Support GCP
+- Distributed write support with dask.array
```
[the remark this answers is not in the document]
Comment (paraseba) [reply docs/docs/contributing.md:74]: Interesting, I thought Deepak's was the ultimate solution we needed.

File: docs/docs/contributing.md [hunk 4]
```diff
@@ -61,42 +59,80 @@ TODO
...
+- Credential sets for virtual datasets
+- Complete Python API
+- Better documentation and examples
+
+#### Performance
+
+- Create scale benchmark of daily updated dataset (15M chunks)
```
[the remark this answers is not in the document]
Comment (paraseba) [reply docs/docs/contributing.md:81]: Going with 30k commits

**Pull request merged**